src/images.py

```diff
@@ -0,0 +1,177 @@
+"""Module for matching individual images from raw and corrected directories
+and computing intensity slopes from the foreground and background regions."""
+import warnings
+
+from pathlib import Path
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+def match_image_pairs(raw_dir, corrected_dir, key_parts=4, extensions=None):
+    """
+    Match images from two directories based on a common key in their filenames.
+
+    Parameters
+    ----------
+    raw_dir : str or Path
+        Directory containing raw images.
+    corrected_dir : str or Path
+        Directory containing corrected images.
+    key_parts : int
+        Number of underscore-separated parts of the filename to use as the key.
+
+        For example, if filenames are named as "AS_1234_A01f00d0.tif", you should
+        set key_parts=3, as the key will be "AS_1234_A01f00d0". The key must include
+        well, site, and channel information to ensure correct pairing.
+    extensions : list or None
+        List of file extensions to include (e.g., ['.tif', '.TIFF']). Default extensions
+        include only versions of TIFF extensions.
+
+    Returns
+    -------
+    List[Tuple[Path, Path]]
+        List of matched (raw_image, corrected_image) pairs.
+    """
+    raw_dir = Path(raw_dir)
+    corrected_dir = Path(corrected_dir)
+    extensions = extensions or [".tif", ".tiff", ".TIFF"]
+
+    raw_files = [
+        f
+        for f in raw_dir.iterdir()
+        if f.suffix.lower() in [e.lower() for e in extensions]
+    ]
+    corrected_files = [
+        f
+        for f in corrected_dir.iterdir()
+        if f.suffix.lower() in [e.lower() for e in extensions]
+    ]
+
+    raw_dict = {"_".join(f.stem.split("_")[:key_parts]): f for f in raw_files}
+    corrected_dict = {
+        "_".join(f.stem.split("_")[:key_parts]): f for f in corrected_files
+    }
+
+    common_keys = set(raw_dict.keys()) & set(corrected_dict.keys())
+    pairs = [(raw_dict[k], corrected_dict[k]) for k in sorted(common_keys)]
+
+    print(f"Found {len(pairs)} image pairs for comparison.")
+    return pairs
+
+def compute_intensity_slopes(image, mask, plot=True):
+    """
+    Compute combined magnitude slope for both foreground and background.
+
+    Parameters
+    ----------
+    image : np.ndarray
+        Input image (2D).
+    mask : np.ndarray (bool)
+        Binary mask (True = foreground).
+    plot : bool
+        If True, make diagnostic plots.
+
+    Returns
+    -------
+    (foreground_mag, background_mag) : tuple of floats
+        Combined slope magnitudes for foreground and background.
```

Comment:
Standard numpy style documentation has the type first and the description at the bottom something like this:


```suggestion
    Returns
    Tuple[float, float]
        A tuple containing two values: the combined slope magnitude for the foreground and the combined slope magnitude for the background.
```

Reply:
Done and done!